client/src/components/forms/SignUp/index.tsx

```diff
@@ -79,6 +140,8 @@ function SignUpForm({
               fullWidth
               value={firstName}
               onChange={(e) => setFirstName(e.target.value)}
+              error={showFirstNameError}
+              helperText={showFirstNameError ? FIRST_NAME_ERROR : undefined}
```

Comment:
(Similar to comment above)
I think all of these could be computed.
E.g.
```
helperText={showFirstNameError(firstName) && FIRST_NAME_ERROR}
```

Reply:
💯 %